### DISCUSSION TITLE
Github copilot stops working after a few minutes.

### DISCUSSION THREAD

**Discussion by anim8rDev** [community/community#45476] [Copilot Conversations] [answered]
I've been having an issue with copilot today and yesterday.  It works fine after launching Visual Studio Code.  However, after a few minutes, it stops giving me suggestions.  As far as I can tell, I'm still connected to copilot.

If I close and re-start visual studio code, it works again for a few minutes then stops again.  Rinse and repeat over and over again.

Thanks

p.s. EDIT:  It is the next day and it won't work at all even after re-start.

**Comment by asolux** [comment]
I have the same problem. Maybe there is a problem with the last update? 
Edit: Also the copilot log output stops.

**Reply by anim8rDev** [reply]
Well.. Misery loves company I guess.   Glad to know it isn't just me.

**Reply by BrettMayson** [reply]
Same thing here, but only on one of my devices so far

**Comment by anim8rDev** [comment]
Copilot stopped working completely.  Re-starts no longer help.  I've logged out and back in to my github account.  I've disabled and re-enabled the extension.  Nothing has worked to this point.

**Reply by jge162** [reply]
Have you tried disabling other extensions. Only turn on copilot. Also you can reset copilot in settings.

**Reply by anim8rDev** [reply]
Unfortunately, that did not help.  I also uninstalled copilot and re-installed it.  It worked for about 3 minutes then stopped.

**Reply by anim8rDev** [reply]
Now the Copilot icon (lower right corner) is spinning continously.

**Reply by jge162** [reply]
You can submit a [bug report](https://support.github.com/contact/bug-report). They can assist you further.

**Reply by jge162** [reply]
> Unfortunately, that did not help. I also uninstalled copilot and re-installed it. It worked for about 3 minutes then stopped.

Your copilot sounds bugged. I recommend you do bug report.

**Reply by anim8rDev** [reply]
Thanks.  I submitted one.  I will report back here if I get a response.

**Reply by asolux** [reply]
For me too, Copilot no longer works on VS Code. Lastly I deleted the "workspaceStorage". Didn't help either.
After each window refresh, Copilot works for a few minutes and then shuts down.
I will also send a bug report.

**Reply by jge162** [reply]
Glad to have helped. Best to you both @anim8rDev @asolux

**Comment by malcolmmurdock** [comment]
Same issue. Works for a little bit. Then stops giving suggestions and I have to restart VSCode.

**Comment by jobo5432** [comment]
Add me to the list. It seems to either stop working almost immediately.

Version Info
=====
Version: 1.74.3 (user setup)
Commit: 97dec172d3256f8ca4bfb2143f3f76b503ca0534
Date: 2023-01-09T16:59:02.252Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: No

Copilot Version: 1.70.8099

**Comment by anim8rDev** [comment]
Those of you having this issue.  Are you developing with Svelte by any chance?  I need to look at this further.  But when Copilot breaks for .svelte files, it may (need to test more) still be working with .js files..

**Reply by anim8rDev** [reply]
It appears that it breaks for .svelte files, but will continue to work on .js and even .json files.  I have disabled all extensions except Copilot.

**Reply by jge162** [reply]
So disabling other extensions have helped? Have you tried resetting copilot in settings Also. That should help too.

**Reply by anim8rDev** [reply]
> So disabling other extensions have helped? Have you tried resetting copilot in settings Also. That should help too.

No, disabling didn't change anything.  I was just noting that I had done that.  When all of this started a few days ago I had been working exclusively in .svelte files.  It wasn't until today when I was working with a .js file that I quickly realized that Copilot was working with those.  I should of checked that on day zero, but I failed.

So right now, I see that while Copilot stops working on .svelte files after a few minutes, it will continue to work on .js files.  I get the same results when all extensions (except copilot) are enabled or disabled.  Resetting copilot doesn't change anything.

Thanks.

**Reply by asolux** [reply]
nope, not for me. I am working on a C# project. I tried to disable all extensions except Copilot, but the problem persists. I opened a ticket and I'm still waiting for a response.

**Reply by jge162** [reply]
Have you check the integrations of your copilot? Maybe look at the GitHub account website side of things. Check copilot setting there.

**Reply by anim8rDev** [reply]
I'm not sure how to check the integrations of Copilot.  Where can I find that?   In my Github acct under Copilot there is nothing really to change other than allowing it to look at public code and use my code.

**Reply by jge162** [reply]
Correct. Try going between allow and block see if that helps.

**Reply by malcolmmurdock** [reply]
I'm not developing in svelte but still have the issue. Vue, Nuxt, Node, and Python for me. 🫤

**Reply by jobo5432** [reply]
Dart/Flutter here, but that includes java and/or kotlin, html, yml, etc... It's just not working.

**Comment by hakula139** [comment]
+1, and I've found an error logged in the developer tools panel of VS Code:

```text
workbench.desktop.main.js:616   ERR Unrecognized language: cpp: Error: Unrecognized language: cpp
    at l (/data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:19934)
    at b (/data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:22921)
    at y (/data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:23147)
    at t.getFunctionPositions (/data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:25129)
    at /data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:50444
    at a.findMatches (/data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:50646)
    at /data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:47119
    at async /data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:47094
    at async /data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:47094
    at async t.getNeighborSnippets (/data/home/hakulachen/.vscode-server/extensions/github.copilot-1.70.8099/dist/worker.js:1:47001)
```

Not sure if it's related to the bug.

**Reply by jge162** [reply]
Unrecognized language. C++, t.getfunction, it doesn’t understand syntax. Hm

**Reply by jge162** [reply]
You should submit that as a [bug report](https://support.github.com/contact/bug-report)

**Reply by hakula139** [reply]
> You should submit that as a bug report.

Thanks, I'll try to submit one.

**Reply by hakula139** [reply]
Done. I'll post here if there's any progress.

**Comment by kiumarsj** [comment]
I also have this problem. Every time i open the vscode after a while i get this error from copilot's extention:
`GitHub Copilot could not connect to server. Extension activation failed: "connect ECONNREFUSED 0.0.0.0:443"`
i tried restarting vscode, restarting my pc, changing network, dns, reinstalling the github copilot and copilot labs extention.
nothing worked.

**Comment by jobo5432** [comment]
This error from the Extension Host Output:
![image](https://user-images.githubusercontent.com/5629514/215343673-b93b5b6a-3ad9-49a9-8b61-f13bff322457.png)

**Reply by asolux** [reply]
Yes. I got the same error under Extension Host output:
```
2023-01-29 21:03:20.464 [error] [GitHub.copilot] provider FAILED
2023-01-29 21:03:20.464 [error] Error: Unrecognized language: csharp
```

Good idea with the nightly build. This extension seems to work. I will test it further tomorrow.

**Comment by jobo5432** [comment]
For what it's worth... I uninstalled the Copilot extension and installed the Copilot Nightly. So far, so good with rendered suggestions.

**Comment by Yonava** [comment]
I am running into the same issue as described in the thread

**Comment by anim8rDev** [comment, marked as answer]
My problem seemed to have been solved.  I received an email response from Support.  It said..

**I believe we had some issues at the end of last week. Could you let me know if it's working again now?**


I'm not ready to say it is completely working, because I've only spent a few minutes in VSCode, but upon first glance, my problems seem to be fixed by whatever they did on their end.  EDIT:: I'm fairly certain it is working for me again 100%.

**Reply by jge162** [reply]
Progress. Glad to hear.

**Comment by kimadeline** [comment]
Hi everyone! For the folks who have been getting `Error: Unrecognized language: ` errors, we are looking into them!

As a workaround for now, using [Copilot nightly](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-nightly) should work, or reverting to version 1.69.0 of the stable extension.

I'll loop back in once there are updates on that bug. So sorry for the inconvenience, and thank you for your patience 🙂

**Reply by RobotKT** [reply]
请问历史版本怎么获取？

**Reply by kimadeline** [reply]
Hi again! Quick update: The issue was caused by an experiment we were running, the experiment has been stopped for now, and we're working on fixing the issue that caused the error in the first place. Any version of Copilot (stable or nightly) should work.

@RobotKT If I understood correctly you're asking how to download an older version of the extension?

In order to do that, open the "Extensions" view, and select "Install Another Version..." in the settings cogwheel for the extension:

![image](https://user-images.githubusercontent.com/51720070/217393116-0272a61d-06ae-45e7-93f3-f57f88060f3b.png)

A dropdown should appear with a list of versions published on the Marketplace:

![image](https://user-images.githubusercontent.com/51720070/217393200-393d9adb-03bc-4eeb-92d4-8cfd3e28b419.png)

**Comment by RobotKT** [comment]
你好，我使用的IDE是vs2022，扩展商店里无法安装历史版本，现在我正在使用版本1.49.0.2，仍然会出现无响应和无法使用，和反应时间很长的现象


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "community/community"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月8日(星期三) 上午7:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [community/community] Github copilot stops working after a few minutes. (Discussion #45476)





 
Hi again! Quick update: The issue was caused by an experiment we were running, the experiment has been stopped for now, and we're working on fixing the issue that caused the error in the first place. Any version of Copilot (stable or nightly) should work.
 
@RobotKT If I understood correctly you're asking how to download an older version of the extension?
 
In order to do that, open the "Extensions" view, and select "Install Another Version..." in the settings cogwheel for the extension:
 

 
A dropdown should appear with a list of versions published on the Marketplace:

**Reply by kimadeline** [reply]
@RobotKT Can you confirm whether you are using Visual Studio, or Visual Studio Code, and the version of the IDE you are using?

**Comment by War3Evo** [comment]
I'm having similar issues.

I estimate I've been using it for 3 weeks now on the 60 day trial.

Debian 11 64 bit ... Visual Studio Code.
GitHub Copilot Extension Released 4/3/2023, 13:16:34

[edit] -- also installed GitHub Copilot for Nightly and that does not solve the issue.

It was working fine for the first 2 weeks I was using it, but just recently it has become non-responsive.  I upgraded to the latest to see if that would solve the issue, and it has not.

Is there any methods I could use to have it do a self diagnostics or something?

**Reply by War3Evo** [reply]
Nightly version 1.80.4:

Here's the log:

```
[INFO] [default] [2023-04-07T18:14:46.068Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-07T18:14:46.326Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-07T18:14:51.933Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 5607 ms
[INFO] [streamChoices] [2023-04-07T18:14:52.016Z] request done: headerRequestId: [57a4a181-d999-4839-a9ac-123039866a81] model deployment ID: [x625ba4f78aa8]
[INFO] [default] [2023-04-07T18:19:15.713Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-07T18:19:16.453Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-04-07T18:19:21.657Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 5204 ms
[INFO] [streamChoices] [2023-04-07T18:19:21.760Z] request done: headerRequestId: [089718d7-9933-497e-8bc4-537021440b6f] model deployment ID: [x625ba4f78aa8]
```

It just doesn't ever show suggestions anymore

**Comment by BaZzz01010101** [comment]
I'm experiencing the same problem last few weeks in Visual Studio. After few minutes of working it just stop suggesting code. The log doesn't have any errors only lines like this:
```
[DEBUG] [getCompletions] [2024-04-01T15:57:26.745Z] Requesting completion at position 15:11, between "      Rock,\r\n      COUNT" and "\r\n    };\r\n".
[INFO] [default] [2024-04-01T15:57:26.899Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-01T15:57:44.301Z] Requesting completion at position 11:16, between "    {\r\n      NONE = -1," and "\r\n      Tree1,\r\n".
[INFO] [default] [2024-04-01T15:57:44.452Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-01T15:57:48.901Z] Requesting completion at position 11:12, between "    {\r\n      Tree1," and "\r\n      Tree2,\r\n".
[INFO] [default] [2024-04-01T15:57:49.010Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-01T15:57:50.746Z] Requesting completion at position 10:5, between "    enum Type : int\r\n    {" and "\r\n      Tree1,\r\n".
[INFO] [default] [2024-04-01T15:57:50.853Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-01T15:57:54.397Z] Requesting completion at position 14:6, between "      Rock,\r\n    };" and "\r\n\r\n".
[INFO] [default] [2024-04-01T15:57:54.518Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```
After restarting Visual Studio it start working but only for a few minutes.

**Comment by Rotarex-Guillaume-Michael** [comment]
Same problem but it looks like my corporate network proxy is involved. It works fine 2-3 minutes and then certificate error which make no sense, if it works once, it should continue to work 😕

![image](https://github.com/community/community/assets/83060543/7a2aa769-f455-45bb-a6f2-2b4445792792)

**Reply by BaZzz01010101** [reply]
I guess you have a different case. In mine there are no errors or warnings at all. It just stop working silently.

**Reply by Rotarex-Guillaume-Michael** [reply]
I started an excel file to find a working version  :D 

(Ongoing testing)
![image](https://github.com/community/community/assets/83060543/efd11f3e-04b3-4178-9677-5970feb344eb)
